src/components/CharacterAI.cpp

```diff
@@ -408,6 +428,24 @@ namespace REGoth
     }
   }
 
+  void CharacterAI::handleFalling()
+  {
+    if (!isInAir)
+    {
+      mFallingVelocity = 0.0f;
+    }
+    else
+    {
+      mFallingVelocity += FALLING_ACCELERATION_Y * bs::gTime().getFixedFrameDelta();
+    }
+
+    if (mVisual->isPlayingFlyingAnimation())
```

Comment:
Shouldn't this be at the beginning of the function or could something go wrong? Other than falling velocity being applied in the same frame? It's a bit confusing to have this at the end. Maybe it also feels confusing as this `if` is not part of *handling* falling but it actually sets the flag which indicate how the falling is handled. Difficult to explain :smile:

Reply:
This is rather an overwrite to let the animation itself take care of the gravity. If I had put at the beginning of the function `FALLING_ACCELERATION_Y` would be added one time, which is not correct. I should probably just rename the function to `handleFallingAndFlying` or so. Do you have a better name? :smile:

Reply:
`handleAirborne`?

Reply:
Oh so it's like two different states `flying` and `falling`? When would you be flying in Gothic? Hm... `handleFallingAndFlying` is a better name :smile:. Is there a way to condense the three `if`s that depend on `isInAir` any further? I find the conditional dependencies (also how `isInAir` differs from `!isStandingOnSolidGround`) still a bit confusing.

Reply:
I added some more information to `isStandingOnSolidGround` explaining what it is. Standing ontop of another dynamic object, such as a another character, is *not* solid ground.

`mIsInAir` just means that the character is not standing on anything. Maybe we can condense those into some functions in a refactoring...

> When would you be flying in Gothic?

Bloodflys do. Also during Jumping, apparently :wink: